src/components/button/Button.tsx

```diff
@@ -0,0 +1,12 @@
+import { ButtonProps } from './Button.props';
+import styles from './Button.module.css';
+
+function Button({ onClick, text, disabled }: ButtonProps) {
+  return (
+    <button className={styles['button']} onClick={onClick} disabled={disabled}>
+      {text}
+    </button>
+  );
+}
```

Comment:
I think you don't need specific types, you can use builtin

```suggestion
import { ButtonHTMLAttributes, DetailedHTMLProps } from 'react';
import styles from './Button.module.css';

function Button({children}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) {
  return (
    <button className={styles['button']}>
      {children}
    </button>
  );
}
```